File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());

        Map<Integer, Integer> map = new HashMap<>();
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            int card = Integer.parseInt(st.nextToken());

            map.put(card, map.getOrDefault(card, 0) + 1);
        }

        int m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < m; i++) {
            int value = Integer.parseInt(st.nextToken());

            if (map.containsKey(value)) {
                sb.append(map.get(value)).append(' ');
            } else {
                sb.append(0).append(' ');
            }
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val sb = StringBuilder()

    val n = br.readLine().toInt()
    val map = HashMap<Int, Int>()

    val st1 = StringTokenizer(br.readLine())
    repeat(n) {
        val num = st1.nextToken().toInt()
        map[num] = (map[num] ?: 0) + 1
    }

    val m = br.readLine().toInt()
    val st2 = StringTokenizer(br.readLine())
    repeat(m) {
        val num = st2.nextToken().toInt()
        sb.append(map[num] ?: 0).append(' ')
    }

    print(sb)
}
